org.adempiere.base/src/org/compiere/model/PO_Record.java

```diff
@@ -109,43 +127,68 @@ static boolean deleteRecordIdCascade (int AD_Table_ID, int Record_ID, String trx
 					}
 					if(count > 0) {
 						if (log.isLoggable(Level.CONFIG)) log.config(table.getName() + " " + s_parentNames[i]  
-								+ " (" + AD_Table_ID + "/" + Record_ID + ") #" + count);
+								+ " (" + AD_Table_ID + "/" + Record_IDorUU + ") #" + count);
 					}
 				}
 			}
 		}
 		return true;
-	}	//	deleteCascade
+	}	//	deleteRecordIdCascade
 
 	//IDEMPIERE-2060
 	/**
 	 * @param tableName
-	 * @param Record_ID
+	 * @param Record_IDorUU record ID (int) or UUID (String)
 	 * @param trxName
 	 */
-	public static void deleteModelCascade(String tableName, int Record_ID, String trxName) {
-		KeyNamePair[] tables = getTablesWithModelCascade(tableName, trxName);
+	public static void deleteModelCascade(String tableName, Serializable Record_IDorUU, String trxName) {
+		int refId;
+		if (Record_IDorUU instanceof Integer) {
+			refId = DisplayType.RecordID;
+		} else if (Record_IDorUU instanceof String) {
+			refId = DisplayType.RecordUU;
+		} else {
+			log.warning(Record_IDorUU.getClass().getName() + " not supported for ID/UUID");
+			return;
+		}
+		KeyNamePair[] tables = getTablesWithModelCascade(refId, tableName, trxName);
 		for (KeyNamePair table : tables) {
 			int dependentTableId = table.getKey();
 			String dependentColumnName = table.getName();
 			String dependentWhere = dependentColumnName + "=?";
 			List<PO> poList = new Query(Env.getCtx(), 
 					MTable.get(dependentTableId).getTableName(),
 					dependentWhere,
-					trxName).setParameters(Record_ID).list();
+					trxName).setParameters(Record_IDorUU).list();
 			for (PO po : poList) {
 				po.deleteEx(true, trxName);
 			}
 		}
 	}
 
 	/**
+	 * @param refId AD_Reference_ID - Record_ID or Record_UU
 	 * @param tableName
 	 * @param trxName
 	 * @return
 	 */
-	private static KeyNamePair[] getTablesWithModelCascade(String tableName, String trxName) {
-		StringBuilder key = new StringBuilder(MColumn.FKCONSTRAINTTYPE_ModelCascade).append("|").append(tableName);
+	private static KeyNamePair[] getTablesWithModelCascade(int refId, String tableName, String trxName) {
+		int refTableDirId;
+		int refTableId;
+		int refTableSearchId;
+		if (refId == DisplayType.RecordID) {
+			refTableDirId = DisplayType.TableDir;
+			refTableId = DisplayType.Table;
+			refTableSearchId = DisplayType.Search;
+		} else if (refId == DisplayType.RecordUU) {
+			refTableDirId = DisplayType.TableDirUU;
+			refTableId = DisplayType.TableUU;
+			refTableSearchId = DisplayType.SearchUU;
+		} else {
+			log.warning(refId + " not supported for ID/UUID");
+			return null;

```

Comment:
better throw illegalargumentexception here. should never happens.

Reply:
done